### DISCUSSION TITLE
parachainStaking provided by substrate itself

### DISCUSSION THREAD

**Discussion by sudipghimire533** [paritytech/substrate#12347] [Ideas]
Can we have something like moonbeam's staking pallet (https://moonbeam.network/announcements/web3-foundation-grant-for-custom-parachain-staking-pallet/) but in substrate.
We can then also integrate it to polkadot UI just like how it is now integrated for collatorSelection pallet.

**Comment by sudipghimire533** [comment]
Basically idea is to have delegated Proof of Stake (dPoS) implementation for collators, designed specifically for Substrate-based parachains.

**Reply by kianenigma** [reply]
This has indeed been discussed and is part of the plan. We almost certainly want to make the staking pallets in substrate more reusable, capable of being configured for both NPoS and DPoS. 

See https://forum.polkadot.network/t/parachain-technical-summit-next-steps/51/17
